client/lib/domains/index.js

```diff
@@ -179,6 +179,12 @@ function hasMappedDomain( domains ) {
 	return getMappedDomains( domains ).length > 0;
 }
 
+function getTld( domain ) {
```

Comment:
I think requiring just the actual domain name (`domain.name`, named `domainName` to keep to the convention) should be enough - then we won't have to do that weird shuffle in https://github.com/Automattic/wp-calypso/pull/10104/files#diff-29c74b95298f4743880af4155d5a2fc2R14 where you have the domain name, you extract the actual domain object from the domains list only to... get the domain name :) I understand that most functions in `lib/domains` operate on the domain object or a list of them, but some of them do work just on `domainName` and I think this one should too.